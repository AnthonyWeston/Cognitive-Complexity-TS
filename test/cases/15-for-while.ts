// increment
function f() {
    while (true);
}

function g() {
    for (;;);
}

function h() {
    do {

    } while (true);
}

// nesting level
function i() {
    // depth 0
    while (true) { // +1 +0
        // depth 1
        i(); // +1 for recursion
    }
}

function j() {
    for (; ;) {
        j();
    }
}

function k() {
    do {
        k();
    } while (true);
}

// nesting increment
function l() {
    // depth 0
    while (true) { // +1
        // depth 1
        while (true); // +1 +1
    }
}

function m() {
    for (; ;) {
        while (true);
    }
}

function n() {
    do {
        while (true);
    } while (true);
}