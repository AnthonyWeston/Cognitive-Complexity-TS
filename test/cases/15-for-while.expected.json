{
    "15-for-while.ts": {
        "score": 20,
        "inner": [
            {
                "name": "f",
                "score": 1,
                "line": 2,
                "column": 1
            },
            {
                "name": "g",
                "score": 1,
                "line": 6,
                "column": 1
            },
            {
                "name": "h",
                "score": 1,
                "line": 10,
                "column": 1
            },
            {
                "name": "i",
                "score": 2,
                "line": 17,
                "column": 1
            },
            {
                "name": "j",
                "score": 2,
                "line": 25,
                "column": 1
            },
            {
                "name": "k",
                "score": 2,
                "line": 31,
                "column": 1
            },
            {
                "name": "l",
                "score": 3,
                "line": 38,
                "column": 1
            },
            {
                "name": "m",
                "score": 3,
                "line": 46,
                "column": 1
            },
            {
                "name": "n",
                "score": 3,
                "line": 52,
                "column": 1
            }
        ]
    }
}